### PR TITLE
Refactor functions internally

### PR DESCRIPTION
**Pull request by catalamarti**

### CODE REVIEW

File: R/summariseInternal.R [hunk 1]
```diff
@@ -0,0 +1,207 @@
+
+summariseCountsInternal <- function(x, strata, counts) {
+  q <- c(
+    'dplyr::n()',
+    'dplyr::n_distinct(.data$person_id)',
+    'dplyr::n_distinct(.data$subject_id)'
+  ) |>
+    rlang::set_names(c("count_records", "count_subjecst", "count_subjects")) |>
+    purrr::keep(c("records", "person_id", "subject_id") %in% counts) |>
```
Comment (cecicampanile) [inline R/summariseInternal.R:9]: is count_subjecst a typo?

Comment (catalamarti) [reply R/summariseInternal.R:9]: Nope, its is just that I allow to be person_id or subject_id but the resulting estimate will be always "number of subjects". At the end the second on dplyr::n_distinct(.data$subject_id) it is not used, so we can delete it if we want

File: R/summariseClinicalRecords.R [hunk 2]
```diff
@@ -71,168 +71,292 @@ summariseClinicalRecords <- function(cdm,
...
-  strata <- c(list(character()), strata)
 
-  # Counts summary ----
-  cli::cli_inform(c("i" = "Summarising {.pkg {omopTableName}} counts and records per person"))
-  result <- summariseRecordsPerPerson(
-    omopTable, date, sex, ageGroup, recordsPerPerson)|>
-    omopgenerics::newSummarisedResult(settings = createSettings(""))
+    # restrict study period
+    omopTable <- restrictStudyPeriod(omopTable, dateRange)
+    if (is.null(omopTable)) return(omopgenerics::emptySummarisedResult())
```
Comment (cecicampanile) [inline R/summariseClinicalRecords.R:142]: should we add the settings to the empty results

Comment (catalamarti) [reply R/summariseClinicalRecords.R:142]: https://github.com/OHDSI/OmopSketch/blob/9c78bf4c84b7710fbf9acaf8def2d330a1451d0d/R/summariseClinicalRecords.R#L273-L274

They are added later when merged with the rest of results, return only affect inside the purrr statement

File: R/summariseConceptSetCounts.R [hunk 3]
```diff
@@ -33,340 +33,175 @@
...
+                                      conceptSet,
+                                      countBy = c("record", "person"),
+                                      concept = TRUE,
+                                      interval = "overall",
+                                      sex = FALSE,
+                                      ageGroup = NULL,
+                                      sample = NULL,
+                                      dateRange = NULL) {
+  # initial check
   omopgenerics::validateCdmArgument(cdm)
```
Comment (cecicampanile) [inline R/summariseConceptSetCounts.R:45]: in some functions this is **cdm <-** omopgenerics::validateCdmArgument(cdm)

**Review by cecicampanile — APPROVED** [review]

**Pull request merged**